Google-Summer-of-Code-2025.md

```diff
@@ -1221,52 +1259,211 @@ This project entails introducing a developer-only UI (as part of the developer o
+  - New fragment, presenter, view model, listeners, and RecyclerView view models + views for the actual platform parameter and feature flag list screens themselves, plus changes for override functionality.
+- Domain changes:
+  - Introduce a new implementation of PlatformParameterController specific to developer options (similar to how HintHandlerDebugImpl exists and is used). This controller:
+    - Delegates functionality to the production implementation of the controller, but overrides as necessary.
+    - Exposes the necessary functionality for displaying platform parameters and feature flags.
+    - Supports local override for syncing by:
+      - Introducing a new persistent cache store for the LocalOverridePlatformParameterDatabase mentioned below).
+      - Introducing a new method for computing the list of ephemeral platform parameters by using both the remote sync and local override, with a resolution order of: pick local override if present, pick remote if present, fallback to default otherwise. This should build on the existing functionality provided by the production implementation.
+    - Introduce an update function for locally overriding platform parameters that changes the new database (but should not affect current in-process state set in PlatformParameterProcessState).
+    - Introduce support for resetting a flag/parameter (which should just remove its entry from the override database).
+- Model changes:
+  - A new LocalOverridePlatformParameterDatabase proto (essentially mirroring the existing RemotePlatformParameterAndFeatureFlagDatabase proto) for storing overridden values. This is kept separate from the other database so that a flag/parameter 'reset' can restore back to the original sync state.
+  - Add 'LOCAL_OVERRIDE' to sync status. This should be used in the override database by default so that the UI is able to properly display the badge for a locally overridden value.
+- Testing hints:
+  - Refer to [this example](https://github.com/oppia/oppia-android/blob/08e00847d955ec1f38fe8000832a2cff1bbff62f/app/src/sharedTest/java/org/oppia/android/app/splash/SplashActivityTest.kt#L1043) to simulate an app restart in tests. Testing should ensure that overridden values are retained across restarts.
 </details>
 
-
 <details>
 <summary>Suggested PM demo points</summary>
 
 - Milestone 1: The new UI correctly shows all platform parameters and feature flags, and their correct value and sync statuses.
 
-- Milestone 2: The new screen fully supports overriding various platform parameters and feature flag values.
+- Milestone 2: The new screen fully supports overriding various platform parameters and feature flag values. This includes demonstrating both overriding user flows in the new UI, and showing the value changes correctly take effect in the app after a prompted (and accepted) app restart.
```

Comment:
I read the second sentence as: demonstrating (a) overriding user flows in the new UI, and (b) showing that value changes correctly take effect in the app. 

Is that what you meant by the "both" part? Or, did it refer to "the two overriding user flows"? If the latter, what are those "two overriding user flows" -- do you mean overriding feature flags and overriding platform parameters? (I initially sort of thought of them as one flow, so just wanted to clarify that's what this meant)

If the last interpretation is correct, then a suggestion is: "This includes demonstrating the user flows for overriding feature flags and overriding platform parameters in the new UI, and showing that the value changes correctly take effect in the app after a prompted (and accepted) app restart."

Reply:
I think the wording intended the former,but I actually would like the latter. Updated to use your wording as it's much clearer and more correct.